File: app/src/main/java/com/suzei/racoon/ui/add/AddActivity.java
package com.suzei.racoon.ui.add;

import android.content.Intent;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.FrameLayout;

import com.google.android.gms.ads.AdRequest;
import com.google.android.gms.ads.AdView;
import com.google.firebase.auth.FirebaseAuth;
import com.suzei.racoon.R;
import com.suzei.racoon.ui.auth.StartActivity;
import com.suzei.racoon.util.OnlineStatus;

import butterknife.BindView;
import butterknife.ButterKnife;

public class AddActivity extends AppCompatActivity {

    public static final String EXTRA_FRAGMENT_TYPE = "fragment_type";

    @BindView(R.id.creator_fragment_container) FrameLayout container;
    @BindView(R.id.creator_banner_ad) AdView bannerAd;

    public static class Add {
        public static final int WORLD = 0;
        public static final int SINGLE_CHAT = 1;
        public static final int GROUP_CHAT = 2;
        public static final int FRIENDS = 3;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_creator);
        ButterKnife.bind(this);
        showFragment();
        setUpBannerAd();
    }

    private void showFragment() {
        int fragmentType = getIntent().getIntExtra(EXTRA_FRAGMENT_TYPE, -1);
        Fragment fragment;
        switch (fragmentType) {

            case Add.WORLD:
                fragment = new AddWorldFragment();
                break;

            case Add.SINGLE_CHAT:
                fragment = new AddSingleChatFragment();
                break;

            case Add.GROUP_CHAT:
                fragment = new AddGroupChatFragment();
                break;

            case Add.FRIENDS:
                fragment = new AddFriendFragment();
                break;

            default:
                throw new IllegalArgumentException("Invalid fragment type=" + fragmentType);
        }

        FragmentManager fm = getSupportFragmentManager();
        FragmentTransaction ft = fm.beginTransaction();
        ft.replace(container.getId(), fragment);
        ft.commit();
    }

    private void setUpBannerAd() {
        AdRequest adRequest = new AdRequest.Builder().build();
        bannerAd.loadAd(adRequest);
    }

    @Override
    protected void onPause() {
        super.onPause();
        OnlineStatus.set(false);
    }

    @Override
    protected void onPostResume() {
        super.onPostResume();
        OnlineStatus.set(true);
    }

    @Override
    protected void onStart() {
        super.onStart();

        if (FirebaseAuth.getInstance().getCurrentUser() == null) {
            Intent intent = new Intent(AddActivity.this, StartActivity.class);
            intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP|Intent.FLAG_ACTIVITY_NEW_TASK);
            startActivity(intent);
            finish();
        }
    }
}

File: app/src/main/java/com/suzei/racoon/ui/add/AddActivity.kt
package com.suzei.racoon.ui.add

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.google.android.gms.ads.AdRequest
import com.google.firebase.auth.FirebaseAuth
import com.suzei.racoon.databinding.ActivityCreatorBinding
import com.suzei.racoon.ui.auth.login.LoginActivity
import com.suzei.racoon.util.OnlineStatus

class AddActivity : AppCompatActivity() {
    private var _binding: ActivityCreatorBinding? = null
    private val binding get() = _binding!!

    object Add {
        const val WORLD = 0
        const val SINGLE_CHAT = 1
        const val GROUP_CHAT = 2
        const val FRIENDS = 3
    }

    companion object {
        const val EXTRA_FRAGMENT_TYPE = "fragment_type"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        _binding = ActivityCreatorBinding.inflate(layoutInflater)
        setContentView(binding.root)
        showFragment()
        setUpBannerAd()
    }

    override fun onPause() {
        super.onPause()
        OnlineStatus.set(false)
    }

    override fun onPostResume() {
        super.onPostResume()
        OnlineStatus.set(true)
    }

    override fun onStart() {
        super.onStart()
        if (FirebaseAuth.getInstance().currentUser == null) {
            val intent = Intent(this@AddActivity, LoginActivity::class.java)
            intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_NEW_TASK)
            startActivity(intent)
            finish()
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        _binding = null
    }

    private fun showFragment() {
        val fragmentType = intent.getIntExtra(EXTRA_FRAGMENT_TYPE, -1)
        val fragment: Fragment = when (fragmentType) {
            Add.WORLD -> AddWorldFragment()
            Add.SINGLE_CHAT -> AddSingleChatFragment()
            Add.GROUP_CHAT -> AddGroupChatFragment()
            Add.FRIENDS -> AddFriendFragment()
            else -> throw IllegalArgumentException("Invalid fragment type=$fragmentType")
        }
        val fm = supportFragmentManager
        val ft = fm.beginTransaction()
        ft.replace(binding.creatorFragmentContainer.id, fragment)
        ft.commit()
    }

    private fun setUpBannerAd() {
        val adRequest = AdRequest.Builder().build()
        binding.creatorBannerAd.loadAd(adRequest)
    }
}